components/webext-storage/src/error.rs

```diff
@@ -49,8 +49,16 @@ pub enum ErrorKind {
     #[error("Error opening database: {0}")]
     OpenDatabaseError(#[from] sql_support::open_database::Error),
 
-    #[error("Sync Payload Error: {0}")]
-    IncomingPayloadError(#[from] sync15::Error),
+    // When trying to close a connection but we aren't the exclusive owner of the containing Arc<>
+    #[error("Other shared references to this connection are alive")]
+    OtherConnectionReferencesExist,
+
+    // When `Weak::upgrade()` returns None.
+    #[error("The storage database has been closed")]
+    WeakReferenceDropped,
```

Comment:
suggestion:

```suggestion
    DatabaseClosed,
```

"Weak reference" is a bit of an implementation detail—and pedantically, I think it's the strong ref that got dropped, not the weak ref 🫢 I think your `#[error(...)]` message is super clear about what happened, though—let's rename the variant to match?